packages/flutter/lib/src/rendering/animated_size.dart

```diff
@@ -0,0 +1,132 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/animation.dart';
+
+import 'box.dart';
+import 'object.dart';
+import 'shifted_box.dart';
+
+/// A render object that can render child size change animations.
+class RenderAnimatedSize extends RenderAligningShiftedBox {
+  /// Creates a render animated size.
+  RenderAnimatedSize({
+    Duration duration,
+    Curve curve,
+    RenderBox child,
+    FractionalOffset alignment: FractionalOffset.center,
+    VoidCallback callback
```

Comment:
When is this callback triggered?  Consider a more descriptive name like `onFoo`, where "foo" explains when this is called.